src/datatype.c

```diff
@@ -238,6 +238,22 @@ STATIC_INLINE void jl_maybe_allocate_singleton_instance(jl_datatype_t *st)
     }
 }
 
+int jl_datatype_isinlinealloc(jl_datatype_t *ty, int pointerfree) JL_NOTSAFEPOINT
+{
+    if (ty->name->mayinlinealloc && ty->layout) {
+        if (ty->layout->npointers > 0) {
+            if (pointerfree)
+                return 0;
+            if (ty->ninitialized != jl_svec_len(ty->types))
```

Comment:
No need to do it now, but it occurs to me `ninitialized` can move to TypeName as well?

Reply:
Good observation. I can make it `n_uninitialized`, then it would be shared. I'll do that once this is merged. I've also thought about removing `->names`, since that is always shared too.

Reply:
`names` isn't shared for NamedTuples. I guess if we wanted we could fetch it out of the parameters in that case though.